app/interactors/images/update_crop_interactor.rb

```diff
@@ -47,7 +47,8 @@ def crop_params
   def update_edition
     updater = Versioning::RevisionUpdater.new(edition.revision, user)
 
-    updater.update_image(image_revision, false)
+    selected = (context.edition.lead_image_revision == context.image_revision)
```

Comment:
I'm still not sure we should do this, as the code in the updater is still faulty. What we're really doing here is working around the fault. I had a little go at making it better, and I think we could do

```
def currently_lead?(image_revision)
  revision.lead_image_revision&.image_id == image_revision.image_id
end
```

The `remove_image` method probably shouldn't be calling `next_lead_image` and could just use the above one to work out what the next lead image should be (either nil, or the current one).

Reply:
It's not really working around the fault as the updater use's two different methods to remove a lead image. Either using `remove_image` if you click on the remove link or by using `update_image` if a user uncheck's the checkbox on the image metadata page. 

Weirdly that's why the code in the updater was put together like that, so changing the `next_lead_image` method to return the image_revision if it's currently lead actually breaks the checkbox way of deselecting a lead image -.-

Reply:
Oh noes. I see what you mean. I hate that checkbox sometimes! Anyway, there's still a smell here that the crop interactor is having to conjure up something that it shouldn't have to care about. This suggests there's something wrong with the updater interface, so I've had a go at spiking a slightly different one that fixes this: https://github.com/alphagov/content-publisher/compare/ben-updater-interface?expand=1.

All that's left would be to update the unit specs to cover the new method. How does it look to you?